staff_features/resources/step_definitions/resource_archival_object_create.rb

```diff
@@ -0,0 +1,194 @@
+  dropdown_items = all('.typeahead.typeahead-long.dropdown-menu')
+  dropdown_items.first.click
+
+  click_on 'Add Date'
+  select 'Single', from: 'archival_object_dates__0__date_type_'
+  fill_in 'archival_object_dates__0__begin_', with: '2000-01-01'
+
+  click_on 'Add Extent'
+  fill_in 'Number', with: @uuid
+  select 'Cassettes', from: 'archival_object_extents__0__extent_type_'
+
+  click_on 'Add Agent Link'
+  select 'Creator', from: 'archival_object_linked_agents__0__role_'
+  fill_in 'archival_object_linked_agents__0__title_', with: "Accession #{@uuid} Agent Title"
+  fill_in 'archival_object_linked_agents__0__relator_', with: 'annotator'
+  dropdown_items = all('.typeahead.typeahead-long.dropdown-menu')
+  dropdown_items.first.click
+  fill_in 'token-input-archival_object_linked_agents__0__ref_', with: 'test_agent'
+  dropdown_items = all('li.token-input-dropdown-item2')
+  dropdown_items.first.click
+
+  click_on 'Add Accession Link'
+  fill_in 'token-input-accession_links_', with: 'test_accession'
+  dropdown_items = all('li.token-input-dropdown-item2')
+  dropdown_items.first.click
+
+  click_on 'Add Subject'
+  fill_in 'token-input-archival_object_subjects__0__ref_', with: 'test_subject'
+  dropdown_items = all('li.token-input-dropdown-item2')
+  dropdown_items.first.click
+
+  click_on 'Add Note'
+  note_type = find('#notes select')
+  note_type.select 'Abstract'
+  fill_in 'archival_object_notes__0__persistent_id_', with: "Persistent ID #{@uuid}"
+  fill_in 'archival_object_notes__0__label_', with: "Label #{@uuid}"
+  check 'archival_object_notes__0__publish_'
+  page.execute_script("$('#archival_object_notes__0__content__0_').data('CodeMirror').setValue('Content #{@uuid}')")
+
+  click_on 'Add External Document'
+  fill_in 'archival_object_external_documents__0__title_', with: "External Document Title #{@uuid}"
+  fill_in 'archival_object_external_documents__0__location_', with: "External Document Location #{@uuid}"
+  check 'archival_object_external_documents__0__publish_'
+
+  click_on 'Add Rights Statement'
+  select 'Copyright', from: 'archival_object_rights_statements__0__rights_type_'
+  fill_in 'archival_object_rights_statements__0__jurisdiction_', with: 'andorra'
+  dropdown_items = all('.typeahead.typeahead-long.dropdown-menu')
+  dropdown_items.first.click
+  fill_in 'archival_object_rights_statements__0__start_date_', with: ORIGINAL_ACCESSION_RIGHTS_STATEMENT_START_DATE
+
+  click_on 'Save'
+  expect(page).to have_text "Archival Object Archival Object 1 #{@uuid} on Resource Resource #{@uuid} created"
+  wait_for_ajax
+end
+
+When 'the user selects the Archival Object' do
+  click_on "Archival Object 1 #{@uuid}"
+end
+
+Then 'the Archival Object with Title {string} is saved as a child of the Resource' do |title|
+  archival_objects = all('#tree-container .table-row', text: title)
+
+  expect(archival_objects.length).to eq 1
+  expect(archival_objects[0][:class]).to include 'indent-level-1 current'
```

Comment:
this looks a little dubious to me. the test claims it's testing that its saved as a child of the Resource but it's just testing the indent level of the element.

Reply:
@balmas the markup associated with the critical "tree" feature is a bit dubious itself which is a big annoyance. Given what's come before the line in question, the CSS selector tells me the archival object is an immediate child of the resource.

Reply:
To be honest I agree. But I couldn't find any other way to have an indicator that the `ArchivalObject` is a child of a `Resource`. I believe that besides the `e2e-tests` there are also `model/controller/entities/etc.` specs that ensure business logic in a more solid way. However, I don't know if this case is covered. Let me know if you have in mind any other way of testing the hierarchy from the UI perspective.

Reply:
Done